### ISSUE TITLE
Does determined AI evaluate a batch or the whole validation dataset?

### ISSUE THREAD

**Issue by scotthuang1989** [determined-ai/determined#2128]
when determiend  run evaluation, does it run it on whole validation dataset or just a batch?

**Comment by neilconway** [comment]
@scotthuang1989 Validation is performed on the entire validation data set. Validation is typically computed on a per-batch basis (and then reduced to produce a single metric for the entire validation set). With PyTorch, it's also possible to compute a single metric over the entire validation set directly. Did that answer your question?

**Comment by scotthuang1989** [comment]
@neilconway  yes.  thanks.

**Comment by scotthuang1989** [comment]
@neilconway  how about train metrics?

**Comment by neilconway** [comment]
@scotthuang1989 Training is performed on one batch of training data at a time; your training metrics are computed for each batch (e.g., in PyTorch, you can return a dict of training metrics as part of each call to `train_batch`).

**Comment by scotthuang1989** [comment]
got it. It make sense as parameter is changing.
